canonical-web-and-design/cookie-policy
# Add the cookie manager to the package

## Done
- Refactor that project structure to be a set of Classes
- Added buttons to the notification
- Added the cookie manager UI modal

## QA
- Pull this branch
- Run `npm install`
- Run `npm run build`
- Run `python -m SimpleHTTPServer`
- Open http://localhost:8000/
- Click "Accept all and visit site" and check the notification closes and the cookie `_cookies_accepted_all` is set
- Refresh the page and see that the notification does not display again
- Remove the cookie and refresh
- Click "Manage your tracker settings" and see it reveals the modal
- Click "Accept recommended settings" button and check it closes and sets the all cookie
- Clear the cookie and refresh again then open the modal manager again
- Click "Save preferences" with both switches on and check it sets the all cookie
- Open the modal again and try a number of combinations and check the correct cookie is set  

## Screenshots
Cookie notification:
![image](https://user-images.githubusercontent.com/1413534/91364515-d2090900-e7f6-11ea-9ea3-b0c2850a84d5.png)

Modal manager:
![image](https://user-images.githubusercontent.com/1413534/91364541-e64d0600-e7f6-11ea-9b2d-d916929c6bec.png)


Fixes https://github.com/canonical-web-and-design/cookie-policy/issues/70
